docs/virtual_machines/net_binding_plugins/macvtap.md

```diff
@@ -0,0 +1,227 @@
+spec:
+  config: '{
+      "cniVersion": "0.3.1",
+      "name": "macvtapnetwork",
+      "type": "macvtap",
+      "mtu": 1500
+    }'
+```
+
+The object should be created in a "default" namespace where all other namespaces
+can access, or, in the same namespace the VMs reside in.
+
+The requested `k8s.v1.cni.cncf.io/resourceName` annotation must point to an
+exposed host interface (via the `lowerDevice` attribute, on the
+`macvtap-deviceplugin-config` `ConfigMap`).
+
+### Feature Gate
+If not already set, add the `NetworkBindingPlugins` FG.
+```
+kubectl patch kubevirts -n kubevirt kubevirt --type=json -p='[{
+  "op": "add",
+  "path": "/spec/configuration/developerConfiguration/featureGates/-",
+  "value": "NetworkBindingPlugins"
+}]'
+```
+
+> **Note**: The specific macvtap plugin has no FG by its own. It is up to the cluster
+> admin to decide if the plugin is to be available in the cluster.
+> The macvtap binding is still in evaluation, use it with care.
+
+### Macvtap Registration
+The macvtap binding plugin configuration needs to be added to the kubevirt CR
+in order to be used by VMs.
+
+To register the macvtap binding, patch the kubevirt CR as follows:
+
+```yaml
+kubectl patch kubevirts -n kubevirt kubevirt --type=json -p='[{"op": "add", "path": "/spec/configuration/network",   "value": {
+            "binding": {
+                "macvtap": {
+                    "domainAttachmentType": "tap"
+                }
+            }
+        }}]'
+```
+
+### VM Macvtap Network Interface
+Set the VM network interface binding name to reference the one defined in the
+kubevirt CR.
+
+```yaml
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-net-binding-macvtap
+  name: vm-net-binding-macvtap
+spec:
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-net-binding-macvtap
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: containerdisk
+          - disk:
+              bus: virtio
+            name: cloudinitdisk
+          interfaces:
+          - name: podnet
+            masquerade: {}
+          - name: hostnetwork
+            binding:
+              name: macvtap
+          rng: {}
+        resources:
+          requests:
+            memory: 1024M
+      networks:
+      - name: podnet
+        pod: {}
+      - name: hostnetwork
+        multus:
+          networkName: macvtapnetwork
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - containerDisk:
+          image: quay.io/kubevirt/fedora-with-test-tooling-container-disk:v1.1.0
+        name: containerdisk
+      - cloudInitNoCloud:
+          networkData: |
+            version: 2
+            ethernets:
+              eth0:
+                dhcp4: true
+        name: cloudinitdisk
+```
+
+The multus `networkName` value should correspond with the name used in
```

Comment:
I would add the same comment about the binding name that is should correspond to the plugin name.

Reply:
Done